rfcs/0094-use-matrix-for-official-chat.md

```diff
@@ -0,0 +1,62 @@
+---
+feature: use-matrix-for-official-chat
+start-date: 2020-05-19
+author: Ryan Mulligan
+shepherd-team:
+shepherd-leader:
+---
+
+# Summary
+[summary]: #summary
+
+Switch from IRC ([Freenode](https://freenode.net/)) to
+[Matrix](https://matrix.org/)
+([#community:nixos.org][1]) as
+our official community chat.
+
+# Motivation
+[motivation]: #motivation
+
+The current official community chat host, Freenode, has lost its
+leadership and its future is uncertain. Many Free Software communities
+are abandoning it, so its network effects may be negative instead of
+positive.
+
+There are currently 1796 members of the Unofficial NixOS Discord
+channel (244 weekly visitors, 95 weekly communicators). Some of them
+might be convinced by the more modern features available on Matrix to
+move their conversations there.
+
+# Detailed design
+[design]: #detailed-design
+
+Update the [NixOS.org community
+page](https://nixos.org/community/index.html) to point to our Matrix
+Space ([#community:nixos.org][1]) that has already been created which you
+can join by visting the [invite
+link][1].
+
+Update any other places that point to Freenode to point to Matrix
+instead.
+
+Port any bots that might be necessary such as:
+
+* nix-channel-monitor
+* github PR events
+* karma bot
+* eval bot
+* factoids bot
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+People are used to Freenode and it might be harder for people to use
```

Comment:
I strongly oppose the phrase *harder* here. Transitioning from one system to another, from one personally established setup to another, is an effort indeed. But that is due to the fact of it being a transition, not because the other system is harder to use.
A good IRC setup also isn't that simple. E.g. I never got around to set up a properly working bouncer.

Alternative wording: People are used to Freenode, and the transition to another ecosystem requires some effort to get adjusted.

Reply:
It is not _purely_ transition cost, though, unless norms very close to IRC ones are established. A stream of short strings is easier, and for some people it is easier to use than something non-homogenous (but not really structured according to structure of conversation)

Reply:
>  A stream of short strings is easier, and for some people it is easier to use than something non-homogenous (but not really structured according to structure of conversation)

From a protocol pov that is true somehow, as long as you'd like to implement that protocol just on a socket. But most additional complexity for simple matrix clients (e.g bots) is abstracted away pretty well, because it is using a well-established technology stack of HTTP and JSON.

Regarding social norms, we might largely stick to the ones from IRC at the start. But over time at least the ones only established due to IRC constraints might change, like the multi-line pastebinning.

Reply:
From communication point of view there is also this ton of questions what people are expected to notice if they are in a conversation. Reactions? Stickers? Whatever else weird feature? Also, is all the communication uniformly short-form?

That's not about HTTPS.

Reply:
```suggestion
People are used to irc and Freenode and it might be a big change for people to use Matrix, which requires different, code wise more complex clients.
```